### PR TITLE
py-setuptools-git-versioning: new package

### PR DESCRIPTION
**Pull request by angus-g**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256.

**Comment by tldahlgren** [comment]
Adding python gurus as reviewers.

File: var/spack/repos/builtin/packages/py-setuptools-git-versioning/package.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    """Use git repo data for building a version number according PEP-440"""
+
+    homepage = "https://setuptools-git-versioning.readthedocs.io/"
+    pypi = "setuptools-git-versioning/setuptools-git-versioning-1.13.3.tar.gz"
+
+    maintainers("angus-g")
+
+    version("1.13.3", sha256="9dfc59a31dcadcae04bcddc50534ccfc07a25a3180ab5cc1b1e3730217971c63")
+
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-setuptools-git-versioning/package.py]:
```suggestion
    depends_on("py-setuptools", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-setuptools-git-versioning/package.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    depends_on("git")
+    depends_on("py-toml", when="python@:3.10", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-setuptools-git-versioning/package.py]:
```suggestion
    depends_on("py-toml@0.10.2:", when="^python@:3.10", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**